### ISSUE TITLE
errors restoring packages on VS2015 Update 3

### ISSUE THREAD

**Issue by james-yin** [JonPSmith/AspNetReactSamples#1]
I just updated my VS2015 to Update 3. The ReactWebPack.CoreRC2 project is having problem restoring packages on my machine. The following is the screenshot (or [http://imgur.com/M1N5Zg8](url)):

![image 1](https://cloud.githubusercontent.com/assets/7233400/16430627/607adcb0-3d31-11e6-8c5a-bb0cb84f9b8a.png)

The other projects loaded fine in the solution.

**Comment by JonPSmith** [comment]
Hi @james-yin,

My first thought is that you haven't installed the [Visual Studio official MSI Installer with Visual Studio tooling](https://www.microsoft.com/net/core#windows) mentioned in the [1. Make sure computer is set up properly](https://github.com/JonPSmith/AspNetReactSamples#1-make-sure-your-computer-is-set-up-properly) of the Readme file. Did you do that? If you did then its some other problem.

In case anyone else reads this then in future this might not work, as Core 1 is released and RC2 may go away. Currently (i.e. on 29th June 2016) the MSI Installer with Visual Studio tooling links to review2 so it should work. However that will change soon.

**Comment by Van-Dame** [comment]
In case installing Visual Studio Tooling doesn't work, try running `dotnet restore` in the project directory from cmdline. Visual Studio sometimes gets hung up on cached packages and doesn't restore from nuget.

**Comment by JonPSmith** [comment]
Thanks @Van-Dame. Helpful suggestion.

**Comment by james-yin** [comment]
@JonPSmith I did install the VS tooling as suggested in the readme.

@Van-Dame I tried `dotnet restore`, but I got a bunch of errors about failed to download packages due to timeouts. I used the VS developer command line and have no problems with other package managers like npm.

`log  : Failed to download package from 'https://api.nuget.org/v3-flatcontainer/m
icrosoft.extensions.configuration.fileextensions/1.0.0-rc2-final/microsoft.exten
sions.configuration.fileextensions.1.0.0-rc2-final.nupkg'.`
`log  : The HTTP request to 'GET https://api.nuget.org/v3-flatcontainer/microsoft
.extensions.configuration.fileextensions/1.0.0-rc2-final/microsoft.extensions.co
nfiguration.fileextensions.1.0.0-rc2-final.nupkg' has timed out after 100000ms.
`

**Comment by JonPSmith** [comment]
Hi @james-yin,

I wanted to rule out that the new Core 1.0.0 hasn't changed anything. I therefore checked that I could see a `...rc2-final.nupkg` for the first packages, `Microsoft.AspNetCore.Diagnostics`, by right-clicking the References directory and selecting `Manage NuGet Packages` and searching for `Microsoft.AspNetCore.Diagnostics` and it is definitely available. 

There have been quite a few clones of this solution and no one else has reported a problem so having ruled out problems cause by the release of Core 1.0.0 then it must be something else.

As @Van-Dame said ASP.Net Core can be funny with restore. I remember having a problem at the start but I can't remember what I did. Maybe try a restart of VS2015 and then do a rebuild on the ReactWebPack.CoreRC2 project.

Sorry I don't have a solution for you. If you are just interested in the build toolchain then you can allways look at the ReactWebPack.MVC5 version.

**Comment by JonPSmith** [comment]
Hi @james-yin,

Did you fix your problem? 

I updated my VS2015 to Update 3 and added the preview 2 tooling update. I then took a new clone of ASpNetReactSamples and did a compile, which forces a restore. The restore of the NuGet packages worked fine for me and [ReactWebPack.CoreRC2](https://github.com/JonPSmith/AspNetReactSamples/tree/master/ReactWebPack.CoreRC2) ran properly.

**Comment by james-yin** [comment]
Hi @JonPSmith ,

Thanks for checking in with me.

I did a fresh clone and tried the build again. The dotnet core stuff got restored correctly, but there's now complaint about "npm - not installed" under the wwwroot folder:

![image 2](https://cloud.githubusercontent.com/assets/7233400/16529108/d990581e-3f75-11e6-9538-118f34aff9c9.png)

I do have the proper node and npm installed on my machine:
![image 3](https://cloud.githubusercontent.com/assets/7233400/16529127/efba9b4a-3f75-11e6-880d-6ce86c5ddca4.png)

The "dev-build" in my Task Runner Explorer for the CoreRC2 project built with no errors, but when I launched the CoreRC2 project with F5, I get errors and exited immediately:
`The program '[14328] dotnet.exe' has exited with code -2147450749 (0x80008083).`
`The program '[5928] iisexpress.exe: Program Trace' has exited with code 0 (0x0).`
`The program '[5928] iisexpress.exe' has exited with code 0 (0x0).`

Any other suggestions please? Thanks.

**Comment by JonPSmith** [comment]
Hi @james-yin,

Glad you got the restore going. 

The `npm - not installed` issue is a [known problem](https://github.com/aspnet/Tooling/issues/479) in the aspnet tooling. I have a note about this in the readme file under the [2. Set up the specific project](https://github.com/JonPSmith/AspNetReactSamples#2-set-up-the-specific-project).

I have no idea on the CoreRC2 exiting on running using F5. I built and ran CoreRC2 successfully in my new cloned version. Have you tried creating a brand new ASP.NET Core web site and trying to run it?

**Comment by JonPSmith** [comment]
Hi @james-yin, Did you get anywhere on getting ReactWebPack.CoreRC2 going?

**Comment by amithegde** [comment]
I get same error:
![npm](https://cloud.githubusercontent.com/assets/816051/16903432/fb7146c2-4c32-11e6-853c-8305a21de730.png)

However, I am able to run the application and it works fine..!

**Comment by JonPSmith** [comment]
Hi @amithegde,

Glad it worked for you. On the `npm - not installed` issue this is a [known problem](https://github.com/aspnet/Tooling/issues/479) in the aspnet tooling and I have note about this in the readme file under the [2. Set up the specific project](https://github.com/JonPSmith/AspNetReactSamples#2-set-up-the-specific-project).

Jon

**Comment by MarkPieszak** [comment]
Same happening here for me with VS2015 update 3. (Windows 7)
`Exception: Call to Node module failed with error: To use prerendering, you must install the 'aspnet-prerendering' NPM package.`

The project will also not run, gets an Internal Server error.

**Comment by JonPSmith** [comment]
Hi @MarkPieszak,

OK, I'm going to need your help on debugging this as I can't reproduce it. I am on a Windows 10 laptop running Visual Studio Professional Update 3 and tried the following:
1. Downloaded a new clone of AspNetReactSamples.
2. Compiled it with F6 (which takes a lot of time as it downloads all the NPM libraries for ReactWebPack.CoreRC2).
3. Set ReactWebPack.CoreRC2 as the startup project.
4. Pressed F5 to start it in debug mode.

That work for me, but it clearly doesn't work for you, and something quite different didn't work for @james-yin (@james-yin - any update on getting it to work or have you just given up on it?). However it did work for @amithegde.

I am contemplating changing  ReactWebPack.CoreRC2 over to RTM 1.0.0, but as I don't know what is going wrong then I'm not even sure that will work. 

Have you any ideas @MarkPieszak? I did notice that a new ASP.NET Core 1.0.0 web project has a entry for `"BundlerMinifier.Core": "2.0.238",` in the tools section of the `project.json` file which isn't in the RC2 version. As your error is about prerendering then it might matter. Can you try adding that and see if it changes anything.

Failing that I could create a branch with ReactWebPack.CoreRC2 changed to RTM 1.0.0 for you to try if you like. Let me know.

**Comment by JonPSmith** [comment]
Hi @MarkPieszak, @james-yin and all watchers,

I have created a new branch called [CoreRC2ToRTM](https://github.com/JonPSmith/AspNetReactSamples/tree/CoreRC2ToRTM) where I have converted the ASP.NET Core project from the RC2 release to the current release, 1.0.0. It works for me, but I would really like you guys to check it and see if it helps sort out your problems. 

Therefore, **could the people who had a problem, and any others watching this project, please check the  [CoreRC2ToRTM](https://github.com/JonPSmith/AspNetReactSamples/tree/CoreRC2ToRTM) branch and let me know if it works.** Once I know it works for you I will merge this into the master branch. Thanks.

---

### List of changes:
1. Change the various NuGet packages from 1.0.0-rc2 to 1.0.0 (plus a [few other changes needed](https://wildermuth.com/2016/06/27/Converting-ASP-NET-Core-1-0-RC2-to-RTM-Bits)).
2. Changed over to using the "Microsoft.AspNetCore.Razor.Tools" version "1.0.0-preview2-final" 
3. Removed Gulp and swapped over to the new Core 1.0.0 Bundler and Minifier
4. Changed the namespace from `ReactWebPack.CoreRC2` to `ReactWebPack.Core`
5. Changed some text in the razor layout files to remove RC2.
6. Changed the Readme file to refer to `ReactWebPack.Core`

_NOTE: I have NOT changed the folder name as that would invalidate all the links I have to the project from my articles._

**Comment by JonPSmith** [comment]
Hi all,

I have combined the CoreRC2ToRTM branch into master so that the [ReactWebPack.Core](AspNetReactSamples/ReactWebPack.CoreRC2/) is now at Core 1.0.0 level. I have also updated the article on my web site http://www.thereformedprogrammer.net to refer to ReactWebPack.Core, not ReactWebPack.CoreRC2.

NOTE: I have changed the project's namespace to **ReactWebPack.Core**, but not the directory name is still ReactWebPack.CoreRC2, as change that causes all sorts of problems to the VS project, and invalidates any links.

I am therefore closing this issue.
